src/main/java/com/example/battleship/game/Ship.java

```diff
@@ -29,13 +29,6 @@ public boolean checkIfSunken(){
         return true;
     }
 
-
-
-    public Map<Field, ShipPart> getShipParts() {
-        return shipParts;
-    }
-    abstract Field getShipField(int i);
```

Comment:
a moze tutaj bys zaimplementowal logike ktora robisz poziom nizej? `getShipParts(int i) { return shipParts.get(i).getPosition()` zamiast implementowac to w kazdym dziecku?

Reply:
this has already been changed, there is no need for getting shipParts from ship childs, nor in Ship mother